### DISCUSSION TITLE
Starlette version requirements when using third party libraries

### DISCUSSION THREAD

**Discussion by kevinvalk** [encode/starlette#1278] [Q&A] [answered]
I am wondering how to go about usage of third party Starlette libraries when building an ASGI application directly on top of Starlette.

Whenever I use any third party libraries/middle like https://github.com/auredentan/starlette-session or https://github.com/mirumee/ariadne, I lock myself to a certain version of Starlette that they used/accept. I fear that in the future, it will be impossible to use libraries because all of them lock to certain Starlette versions that turn out to be incompatible in requirements (this may or may not be an actual incompatibility if you talk about breaking changes in versions).

Is there any recommendation, guide, best practice regarding versioning of Starlette and using it to build something. Perhaps on both the library and end consumer side.

**Comment by Kludex** [comment, marked as answer]
I'd recommend to not lock Starlette version. We do our best to not break any project that uses Starlette.

The last relevant breaking change was the change to the `TestClient`, which only breaks the test suite, and not the application itself.

**Reply by Kludex** [reply]
Also, the two mentioned projects don't lock the version anymore. :pray:
